### PR TITLE
FINALLY fix mag rotation issues.

### PR DESCRIPTION
**Pull request by mhkabir**
Fixes https://github.com/PX4/Firmware/issues/7358.

### CODE REVIEW

**Comment by mhkabir** [comment]
@matanhavi @tubme please test this PR thoroughly with all the problem cases you found.

@LorenzMeier I also realise that we can cut our loop times for voted_sensors_update in half (because I made small optimisations over the last few PRs which remove bad assumptions preventing this previously). We are currently super-wasteful with nested loops in the code. Something to remember after this fix goes in.

**Comment by mhkabir** [comment]
@bresch please test this :)

**Comment by matanhavi** [comment]
I tested with the same configuration, and the PR fixes the issue.
Note that I tested it only on the table.

File: src/modules/sensors/voted_sensors_update.cpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ VotedSensorsUpdate::VotedSensorsUpdate(const Parameters &parameters, bool hil_en
 	// initialise the corrections
 	memset(&_corrections, 0, sizeof(_corrections));
 
-	for (unsigned i = 0; i < 3; i++) {
+	for (unsigned i = 0; i < math::max(ACCEL_COUNT_MAX, GYRO_COUNT_MAX); i++) {
```
Comment (bkueng) [inline src/modules/sensors/voted_sensors_update.cpp]: This is wrong: `i` stands for the axis index, not sensor index.

Comment (mhkabir) [inline src/modules/sensors/voted_sensors_update.cpp]: Fixed.

**Comment by tubeme** [comment]
This does not fix our issue. On the X2.1 the external mag0 was 180 degrees offset (the GPS + Mag physically are oriented toward the front). Even more, The CAL_MAG0_ROT was at Internal Mag (-1) and I could not change it! I tried a couple of times to give Yaw 180 degrees without success. I tried forcefully to change it with param 4 but again it did not store my choice. It seems like the QGC accepts the change but then it stays back to Internal Mag. I know there are 3 internal mags in X2.1 and 1 external. I don't see a MAG3 in Parameters...

**Comment by mhkabir** [comment]
@tubme can you give us a parameter dump? Please post it here : https://github.com/PX4/Firmware/pull/7434 (instead of on this closed PR).

**Pull request merged**